### PR TITLE
Lddl bert

### PR DESCRIPTION
**Pull request by wdykas**
# What does this PR do ?

This adds the ability to use LDDL dataloaders in Nemo FW

**Collection**: [NLP, NemoFW

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage

add LDDL as dataloader in config. Instructions will be added to the Megatron Launcher for data prep and use.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
-  New Feature

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_bert_model.py [hunk 1]
```diff
@@ -49,6 +50,14 @@
 
     HAVE_APEX = False
 
+try:
+    import logging
+    from lddl.torch_mp import get_bert_pretrain_data_loader
```
Comment (shanmugamr1992) [inline nemo/collections/nlp/models/language_modeling/megatron_bert_model.py:55]: Can  we add the library to requirements file in Nemo. @ericharper

Comment (shanmugamr1992) [reply nemo/collections/nlp/models/language_modeling/megatron_bert_model.py:55]: @wdykas

Comment (ericharper) [reply nemo/collections/nlp/models/language_modeling/megatron_bert_model.py:55]: yes, add it to requirements_nlp.txt

Comment (wdykas) [reply nemo/collections/nlp/models/language_modeling/megatron_bert_model.py:55]: done

File: requirements/requirements_nlp.txt [hunk 2]
```diff
@@ -21,3 +21,4 @@ rapidfuzz
 rouge_score
 sacrebleu[ja]
 sentence_transformers
+git+https://github.com/NVIDIA/lddl.git
```
Comment (ericharper) [inline requirements/requirements_nlp.txt:24]: There's no pip install?

Comment (wdykas) [reply requirements/requirements_nlp.txt]: this is how the pip install is specified in the main package https://github.com/NVIDIA/LDDL.

File: requirements/requirements_nlp.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@ fasttext
 flask_restful
 ftfy
 gdown
+git+https://github.com/NVIDIA/lddl.git
```
Comment (ericharper) [inline requirements/requirements_nlp.txt:8]: Remove and make optional until we have a pip package.

**Review by ericharper — CHANGES_REQUESTED** [review]

File: requirements/requirements_nlp.txt (hunk 3 above)
Comment (wdykas) [reply requirements/requirements_nlp.txt]: removed import and added instructions to launcher for optional import

**Review by ericharper — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**